pkg/plugin/v3/scaffolds/internal/templates/main.go

```diff
@@ -228,13 +230,26 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+{{ if not .ComponentConfig }}
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:             scheme,
 		MetricsBindAddress: metricsAddr,
 		Port:               9443,
 		LeaderElection:     enableLeaderElection,
 		LeaderElectionID:   "{{ hashFNV .Repo }}.{{ .Domain }}",
 	})
+{{- else }}
+	options, err := ctrl.Options{
+		Scheme:         scheme,
+		LeaderElection: enableLeaderElection,
+	}.AndFrom(ctrl.ConfigFile().AtPath("config/manager/controller_manager_config.yaml"))
```

Comment:
Nit: set `const managerConfigPath = "config/manager/controller_manager_config.yaml"` at the top of the file and use variable here

Reply:
I would say that, to keep it similar to the rest of the files, not using a `const` in this case is ok. If we want them to be `const` we should probably do it for all the templates and not only this one, and that would be a separate clean up PR.

Reply:
I actually think I should be making this a flag `--config` to follow the standard practice with component config implementers.

Reply:
^ sgtm